src/components/HomePage.js

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { fetchData } from '../redux/weather/weathersReducer';
+import chooseImage from './ImageFunction';
+import './style/homePage.css';
+import microphone from '../Image/green-radio-microphone-14640.png';
+import setting from '../Image/settings-5666.png';
+import rightIcon from '../Image/right-arrow-6421.png';
+
+const ShowAll = () => {
+  const dispatch = useDispatch();
+  const weatherData = useSelector((state) => state.weathers.weatherData);
+  const [selectedOption, setSelectedOption] = useState('All'); // State to store the selected option value
+
+  useEffect(() => {
+    dispatch(fetchData());
+  }, [dispatch]);
+  const filteredData = weatherData.filter((city) => {
+    if (selectedOption === 'All') {
+      return true; // Show all cities when 'All' is selected
+    }
+    if (selectedOption === 'Hot') {
+      return Math.floor(city.main.temp - 273.15) > 21;
+      // Filter based on temperature greater than 21
+    }
+    if (selectedOption === 'Cold') {
+      return Math.floor(city.main.temp - 273.15) < 10; // Filter based on temperature less than 10
+    }
+    if (selectedOption === 'Temperate') {
+      return Math.floor(city.main.temp - 273.15) >= 10 && Math.floor(city.main.temp - 273.15) <= 21;
+      // Filter based on temperature between 10 and 21
+    }
+    return true;
+  });
```

Comment:
- [OPTIONAL] I would strongly recommend the incorporation of a search bar filter function within the project to enhance the overall user experience. This feature would empower users to filter cities based on their names, providing a more intuitive and user-friendly way to navigate and access the desired information. Such an addition would not only align with user expectations but also contribute significantly to the project's usability and accessibility, ultimately resulting in a more satisfying experience for the end-users.